webapp/src/js/services/db-sync.js

```diff
@@ -73,20 +75,34 @@ angular
             return;
           }
           const remote = DB({ remote: true });
+          const onDeniedPromises = [];
           return DB()
             .replicate[direction.name](remote, direction.options)
             .on('denied', function(err) {
-              // In theory this could be caused by 401s
-              // TODO: work out what `err` looks like and navigate to login
-              // when we detect it's a 401
```

Comment:
The 401 would emit an `error`, not a `denied` event. 
According to CouchDB documentation (http://docs.couchdb.org/en/2.2.0/api/database/bulk-api.html#api-db-bulk-docs-validation) the only error types possible here are "conflict" (which we never receive because of `new_edits`) and "forbidden" which we are now handling. 
We already watch for `401` status codes in the custom fetch function: https://github.com/medic/cht-core/blob/master/webapp/src/js/controllers/inbox.js#L166